### PR TITLE
Feature/backend init

### PR DESCRIPTION
**Pull request by amirfakhrullah**
- install basic packages for backend usage
 - setup `mongodb` (need to add the env values in heroku backend before merge this PR)
 - setup `passport.js` for authentication

### CODE REVIEW

File: server/src/index.ts [hunk 1]
```diff
@@ -1,16 +1,69 @@
...
+
+// middlewares
+app.use(
+  cors({
+    origin: IS_PROD ? FRONTEND_URL : "*",
+    credentials: true,
+  })
+);
+app.use(helmet());
+app.use(compression());
```
Comment (abdulqshabbir) [inline server/src/index.ts:29]: For the compression package are we compressing all request bodies coming to the server? Or should we create a filter function?
Not to sure here but just want your take on it :)

Comment (amirfakhrullah) [reply server/src/index.ts:29]: Yes, we can add a filter function in there. But for now I think we can compress everything first and see how it goes. We can add a filter function if needed later on.

What do you think?

Comment (abdulqshabbir) [reply server/src/index.ts:29]: Yeah that sounds good for now :)

**Review by abdulqshabbir — APPROVED** [review]
Looks good!

**Review by cherylli — APPROVED** [review]
works perfectly with the env variables added locally

**Review by AllMikeNoIke — COMMENTED** [review]
I think it would be a lot simpler to use JWT's for authorization, instead of Passport, considering we're not implementing third-party sign in for MVP.

**Pull request merged**